src/frontend/src/btc/utils/btc-transactions.utils.ts

```diff
@@ -16,11 +16,51 @@ export const mapBtcTransaction = ({
 	btcAddress: BtcAddress;
 	latestBitcoinBlockHeight: number;
 }): BtcTransactionUi => {
-	const isTypeSend = inputs.some(({ prev_out }) => prev_out.addr === btcAddress);
-	const output: BitcoinOutput | undefined = out.find(({ addr }) =>
-		isTypeSend ? addr !== btcAddress : addr === btcAddress
+	const { totalInputValue, isTypeSend } = inputs.reduce<{
+		totalInputValue: number;
+		isTypeSend: boolean;
+	}>(
+		(acc, { prev_out: { value, addr } }) => ({
+			totalInputValue: acc.totalInputValue + value,
+			isTypeSend: !acc.isTypeSend ? addr === btcAddress : acc.isTypeSend
+		}),
+		{
+			totalInputValue: 0,
+			isTypeSend: false
+		}
 	);
 
+	const { totalOutputValue, value, to } = out.reduce<{
+		totalOutputValue: number;
+		value: number | undefined;
+		to: string | undefined;
+	}>(
+		(acc, { addr, value }) => {
+			const isValidOutput =
+				(isTypeSend && addr !== btcAddress) || (!isTypeSend && addr === btcAddress);
```

Comment:
I was wondering in which case it would appear nullish `value`: what happens if, for example, I stupidly send BTC to myself? what will be `value`? My understanding from the code is that it stays `undefined` because there will be no valid output. If yes, is that on purpose?

P.S. same question for `to` address

Reply:
Good question, I think we'll need to test this kind of corner cases, because I'm not sure how the Blockchain API response will look like in the provided example. Not sure how it's done for other tokens, but maybe it makes sense to disallow sending to the current address?

Reply:
I would not disallow completely, maybe just a warning... I could be used as test, to send to itself, or something similar.

But about the case, maybe try to send to your self something as a test? Just to check? And if you don't want to tackle it in this one, add a TODO?

Reply:
@AntonioVentilii-DFINITY you're right, value is undefined when transaction's sender and receiver addresses are the same:

<img width="664" alt="Screenshot 2024-10-15 at 11 31 14" src="https://github.com/user-attachments/assets/8bb50682-13a9-4096-a777-87959b956c3c">

Since we already have TODO in the code, I suggest we handle this corner case separately. WDYT?

Reply:
For me all good! Hopefully people don't have BTC to waste like that eheh